src/modules/alunosAdmFormValidation.ts

```diff
@@ -272,3 +266,23 @@ export const handleChangeDepartamento = (
     }
   }
 };
+
+export const handleChangeFilterCpf = (cpf: any, setCpf: any) => {
+  handleChangeNoWhiteSpaceInput(cpf, setCpf);
+
+  if (!validateCpf(cpf)) {
+    showErrorToastCpf(cpf, setCpf);
+  }
+};
+
+const showErrorToastCpf = (cpf: any, setCpf: any) => {
+  if (!!cpf && cpf.length > CPF_LENGTH) {
+    const cpfWithMaxLength = cpf.slice(0, CPF_LENGTH);
+    setCpf(cpfWithMaxLength);
+    console.log(`Apenas ${CPF_LENGTH} caracteres`);
+  } else if (!validateOnlyNumbers(cpf) && !!cpf) {
+    const cpfOnlyNumbers = cpf.slice(0, -1);
+    setCpf(cpfOnlyNumbers);
```

Comment:
// formValidationUtils.ts

Poderia fazer uma correção parecida com:

```
const validateOnlyNumbers = (input: any) => {
  const regex = new RegExp(/^\d+$/); // ou separar essa regex
  const isNumber = regex.test(input);

  return isNumber;
}
```

// handleChangeCpf

```
const isNumber = validateOnlyNumbers(cpf);

if (!isNumber) {
  console.log("Digite apenas números!");

  return;
}

if (cpf.length > CPF_LENGTH) {
  console.log(`Digite apenas ${CPF_LENGTH} caracteres!`);

  return;
}

setCpf(cpf);

const fieldKey = "cpf";

if (cpf.length < CPF_LENGTH) {
  const messageObject = { cpf: "Digite o CPF corretamente" };

  updateErrorMessages(setErrorMessages, fieldKey, messageObject);
}

if (cpf.length === CPF_LENGTH) {
  cleanErrorMessages(setErrorMessages, fieldKey);
}
```

`!isNumber` também impede espaços em branco.

`setCpf(cpf);` foi executado apenas uma vez. Se essa função gravou o valor da variável de estado, então significa que o _input_ é um número e não é maior que CPF_LENGTH.

Observe que, dessa forma, a validação está sendo feita na própria função `handleChangeCpf`.

Por que não fazer uma função de validação que retorna mensagens de erro? Afinal, poderia ser reutilizada para a validação do campo cpf do filtro. A resposta é que há mensagens para o usuário visualizar durante a digitação e há outras mensagens que devem ser gravadas em `setErrorMessages`.


Sobre o seu código:

Observe que a função `validateCpf` está dentro do _onChange_. Isso significa que essa função é chamada, no mínimo, onze vezes! Já sabemos que nas dez primeiras vezes é impossível a função retornar `true`, uma vez que a função espera o tamanho `11`. É como se as invocações anteriores fossem inúteis.

```
const validateCpf = (valor: string) => {
  const onlyNumbers = validateOnlyNumbers(valor);
  const elevenNumbers = valor.length === CPF_LENGTH;

  return onlyNumbers && elevenNumbers;
};
```

Observe que a sugestão que fiz evita trechos de código como:

`!!cpf && cpf.length > CPF_LENGTH) || !validateOnlyNumbers(cpf)`